diff -ur kcm-grub2-0.6.4.languages/src/kcm_grub2.cpp kcm-grub2-0.6.4.set_default/src/kcm_grub2.cpp
--- kcm-grub2-0.6.4.languages/src/kcm_grub2.cpp	2016-04-08 12:34:26.541904595 +0100
+++ kcm-grub2-0.6.4.set_default/src/kcm_grub2.cpp	2016-04-08 13:22:04.889249249 +0100
@@ -27,6 +27,7 @@
 #include <QTreeView>
 #include <QtGlobal>
 #include <QtAlgorithms>
+#include <QStandardItemModel>
 
 //KDE
 #include <KAboutData>
@@ -491,8 +492,8 @@
 
     Action saveAction("org.kde.kcontrol.kcmgrub2.save");
     saveAction.setHelperID("org.kde.kcontrol.kcmgrub2");
-    saveAction.addArgument("rawConfigFileContents", configFileContents.toLocal8Bit());
-    saveAction.addArgument("rawDefaultEntry", !m_entries.isEmpty() ? grubDefault : m_settings.value("GRUB_DEFAULT").toLocal8Bit());
+    saveAction.addArgument("rawConfigFileContents", configFileContents.toUtf8());
+    saveAction.addArgument("rawDefaultEntry", !m_entries.isEmpty() ? grubDefault : m_settings.value("GRUB_DEFAULT").toUtf8());
     if (m_dirtyBits.testBit(memtestDirty)) {
         saveAction.addArgument("memtest", ui->checkBox_memtest->isChecked());
     }
@@ -522,7 +523,7 @@
         dialog->setModal(true);
         dialog->setDefaultButton(KDialog::Ok);
         dialog->setEscapeButton(KDialog::Ok);
-        KMessageBox::createKMessageBox(dialog, QMessageBox::Information, i18nc("@info", "Successfully saved GRUB settings."), QStringList(), QString(), 0, KMessageBox::Notify, reply.data().value("output").toString()); // krazy:exclude=qclasses
+        KMessageBox::createKMessageBox(dialog, QMessageBox::Information, i18nc("@info", "Successfully saved GRUB settings."), QStringList(), QString(), 0, KMessageBox::Notify, QString::fromUtf8(reply.data().value("output").toByteArray())); // krazy:exclude=qclasses
         load();
     } else {
         KMessageBox::detailedError(this, i18nc("@info", "Failed to save GRUB settings."), reply.errorDescription());
@@ -1245,7 +1246,7 @@
         errorMessage = i18nc("@info", "The process crashed.");
         break;
     default:
-        errorMessage = QString::fromLocal8Bit(reply.data().value(QLatin1String("output")).toByteArray());
+        errorMessage = QString::fromUtf8(reply.data().value(QLatin1String("output")).toByteArray());
         break;
     }
     reply.addData(QLatin1String("errorMessage"), errorMessage);
